src/api/object.js

```diff
@@ -1,15 +1,71 @@
 'use strict'
 
 const argCommand = require('../cmd-helpers').argCommand
+const DAGNode = require('ipfs-merkle-dag').DAGNode
 
 module.exports = (send) => {
   return {
     get: argCommand(send, 'object/get'),
-    put (file, encoding, cb) {
-      if (typeof encoding === 'function') {
-        return cb(null, new Error("Must specify an object encoding ('json' or 'protobuf')"))
+    put (obj, options, callback) {
+      if (typeof options === 'function') {
+        callback = options
+        options = {}
       }
-      return send('object/put', encoding, null, file, cb)
+      if (!options) {
+        options = {}
+      }
+
+      let tmpObj = {
+        Data: null,
+        Links: []
+      }
+
+      if (Buffer.isBuffer(obj)) {
+        if (!options.enc) {
+          tmpObj = { Data: obj.toString(), Links: [] }
+        } else {
+          tmpObj = JSON.parse(obj.toString())
```

Comment:
wait no, we actually don't want to parse at all in here, that's the job of ipfs itself isn't it?